sycl/plugins/unified_runtime/ur/adapters/cuda/image.cpp

```diff
@@ -0,0 +1,1029 @@
+        RetErr = UR_CHECK_ERROR(
+            cuMemcpyAtoHAsync(dst_with_offset, (CUarray)pSrc, src_offset_bytes,
+                              CopyExtentBytes, Stream));
+      } else if (pImageDesc->type == UR_MEM_TYPE_IMAGE2D) {
+        CUDA_MEMCPY2D cpy_desc = {};
+        cpy_desc.srcXInBytes = srcOffset.x;
+        cpy_desc.srcY = srcOffset.y;
+        cpy_desc.dstXInBytes = dstOffset.x;
+        cpy_desc.dstY = dstOffset.y;
+        cpy_desc.dstMemoryType = CUmemorytype_enum::CU_MEMORYTYPE_HOST;
+        cpy_desc.dstHost = pDst;
+        if (pImageDesc->rowPitch == 0) {
+          cpy_desc.srcMemoryType = CUmemorytype_enum::CU_MEMORYTYPE_ARRAY;
+          cpy_desc.srcArray = (CUarray)pSrc;
+        } else {
+          // Pitched memory
+          cpy_desc.srcMemoryType = CUmemorytype_enum::CU_MEMORYTYPE_DEVICE;
+          cpy_desc.srcPitch = pImageDesc->rowPitch;
+          cpy_desc.srcDevice = (CUdeviceptr)pSrc;
+        }
+        cpy_desc.WidthInBytes = PixelSizeBytes * copyExtent.width;
+        cpy_desc.Height = copyExtent.height;
+        RetErr = UR_CHECK_ERROR(cuMemcpy2DAsync(&cpy_desc, Stream));
+      } else if (pImageDesc->type == UR_MEM_TYPE_IMAGE3D) {
+        CUDA_MEMCPY3D cpy_desc = {};
+        cpy_desc.srcXInBytes = srcOffset.x;
+        cpy_desc.srcY = srcOffset.y;
+        cpy_desc.srcZ = srcOffset.z;
+        cpy_desc.dstXInBytes = dstOffset.x;
+        cpy_desc.dstY = dstOffset.y;
+        cpy_desc.dstZ = dstOffset.z;
+        cpy_desc.srcMemoryType = CUmemorytype_enum::CU_MEMORYTYPE_ARRAY;
+        cpy_desc.srcArray = (CUarray)pSrc;
+        cpy_desc.dstMemoryType = CUmemorytype_enum::CU_MEMORYTYPE_HOST;
+        cpy_desc.dstHost = pDst;
+        cpy_desc.WidthInBytes = PixelSizeBytes * copyExtent.width;
+        cpy_desc.Height = copyExtent.height;
+        cpy_desc.Depth = copyExtent.depth;
+        RetErr = UR_CHECK_ERROR(cuMemcpy3DAsync(&cpy_desc, Stream));
+      }
+    } else {
+      /// imageCopyFlags == UR_EXP_IMAGE_COPY_FLAG_DEVICE_TO_DEVICE
+      /// TODO: implemet device to device copy
+      return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+    }
+    if (phEvent) {
+      auto NewEvent = ur_event_handle_t_::makeNative(UR_COMMAND_MEM_IMAGE_COPY,
+                                                     hQueue, Stream);
+      NewEvent->record();
+      *phEvent = NewEvent;
+    }
+  } catch (ur_result_t Err) {
+    return Err;
+  } catch (...) {
+    return UR_RESULT_ERROR_UNKNOWN;
+  }
+
+  return RetErr;
+}
+
+UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesImageGetInfoExp(
+    ur_exp_image_mem_handle_t hImageMem, ur_image_info_t propName,
+    void *pPropValue, size_t *pPropSizeRet) {
+  UR_ASSERT(hImageMem, UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+
+  CUDA_ARRAY3D_DESCRIPTOR ArrayDesc;
+  ur_result_t RetErr =
+      UR_CHECK_ERROR(cuArray3DGetDescriptor(&ArrayDesc, (CUarray)hImageMem));
+  switch (propName) {
+  case UR_IMAGE_INFO_WIDTH:
+    if (pPropValue)
```

Comment:
This patch, even within this file, has a mixture of curly braces for single statement `if`s,
https://github.com/intel/llvm/pull/10112/files/244497b69279253d1af21161fa406b31e429f572#diff-a42388f0f22ea1ea028ca05e1126fe8c1ea2d6fa7b4e1260531d1ccf0e873078R422 vs: https://github.com/intel/llvm/pull/10112/files/244497b69279253d1af21161fa406b31e429f572#diff-a42388f0f22ea1ea028ca05e1126fe8c1ea2d6fa7b4e1260531d1ccf0e873078R706
would be good to align on that.

Reply:
It seems to be the norm to exclude the brace in `pi2ur.hpp`. There is a mixture within this CUDA adaptor directory (even within other files within this directory).

I've tidied this file to opt for using the braces, and left `pi2ur.hpp` omitting them. I'll leave the rest.